nifty-ssl/src/main/java/com/facebook/nifty/ssl/PollingMultiFileWatcher.java

```diff
@@ -0,0 +1,275 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.google.common.io.BaseEncoding;
+import com.google.common.io.Files;
+import com.google.common.util.concurrent.ListenableScheduledFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.airlift.log.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An implementation of {@link MultiFileWatcher} that polls the watched files for changes at regular intervals
+ * in a background thread.
+ */
+public final class PollingMultiFileWatcher implements MultiFileWatcher {
+    // Fields set in constructor
+    private final long initialDelay;
+    private final long interval;
+    private final TimeUnit timeUnit;
+    // Fields set in registerUpdateCallback()
+    private Consumer<Set<File>> callback;
+    private Set<File> watchedFiles;
+    // Fields set in start()
+    private ListeningScheduledExecutorService executorService;
+    private ListenableScheduledFuture<?> future;
+    // Cache of last known file state, updated at every polling cycle
+    private final AtomicReference<Map<File, FileMetadata>> metadataCacheRef;
+
+    private static final Logger log = Logger.get(PollingMultiFileWatcher.class);
+
+    /**
+     * Creates a new watcher. The watcher doesn't start scanning files on disk until the {@code start()} method is
+     * called.
+     *
+     * @param initialDelay how long to wait until the first scan of the files.
+     * @param interval how often to rescan the files.
+     * @param timeUnit time unit for {@code initialDelay} and {@code interval}.
+     */
+    public PollingMultiFileWatcher(long initialDelay, long interval, TimeUnit timeUnit) {
+        this.initialDelay = initialDelay;
```

Comment:
When you first initialize the server you will read the ticket seed and key/cert files and create initial configuration from that. So if we're doing a 5 minute polling interval, it could make sense to do intialDelay = interval = 5 minutes. That way you don't do two reads back-to-back.